cpp/.clang-format

```diff
@@ -0,0 +1,156 @@
+---
```

Comment:
Is this the same as what's in cudf, and if so, do we have a way to use it directly rather than copy?

`wget` comes to mind, but I know that could present other problems. :man_shrugging: